tests.py

```diff
@@ -163,6 +163,25 @@ def gen():
     assert list(gen()) == [1, 2, 3]
 
 
+def test_exception_before_first_yield():
+    """Test an exception raised before the first `yield`."""
+
+    def raise_value_error():
+        raise ValueError()
+
+    def subgen():
+        yield raise_value_error()
+
+    @yieldfrom
+    def gen():
+        try:
+            yield From(subgen())
+        except ValueError:
+            yield 1
```

Comment:
Now I see what you meant.
In our example we raised the exception in the except clause instead of yielding something.